apps/cowswap-frontend/src/modules/usdAmount/hooks/useUsdPrice.ts

```diff
@@ -30,5 +30,9 @@ export function useUsdPrice(currency: Nullish<Token>): UsdPriceState | null {
 
   if (!currencyAddress) return null
 
-  return usdPrices[currencyAddress] || null
+  const price = usdPrices[currencyAddress]
+
+  if (price && price.currency.chainId !== currency?.chainId) return null
+
+  return price || null
```

Comment:
How about:

```suggestion
  if (!price || price.currency.chainId !== currency?.chainId) return null

  return price
```

Reply:
If we want to be extra compact:

```suggestion
  return !price || price.currency.chainId !== currency?.chainId ? null : price
```

But that might be less readable...

Reply:
Fixed: https://github.com/cowprotocol/cowswap/pull/3756/commits/08afda61eca55c270ff0dac10aa8a68c5a88ef6d